crates/graphql-composition/src/emit_federated_graph.rs

```diff
@@ -46,80 +52,83 @@ pub(crate) fn emit_federated_graph(mut ir: CompositionIr, subgraphs: &Subgraphs)
     ir.fields
         .sort_unstable_by_key(|field| (field.parent_definition_name, &ctx[field.field_name]));
 
+    emit_directive_definitions(&ir, &mut ctx);
     emit_subgraphs(&mut ctx);
     emit_input_value_definitions(&ir.input_value_definitions, &mut ctx);
     emit_fields(&ir.fields, &mut ctx);
 
     emit_union_members_after_objects(&ir.union_members, &mut ctx);
+    federation_builtins::emit_federation_builtins(&mut ctx);
+
+    emit_directives_and_implements_interface(&mut ctx, ir);
 
-    emit_directives_and_implements_interface(ctx, ir);
+    ctx.out.enum_values.sort_unstable_by_key(|v| v.enum_id);
```

Comment:
You're sure about keeping `ctx`? If I remember correctly I was taking ownership over `ctx` because any modifications after this method wouldn't be correct

Reply:
IR is being `mem::take()`n from but I haven't seen anything problematic with the context. Will double check. I ended up putting the emit_cost_directive() and emit_list_size_directive() calls inside, so emit_directives_and_implements_interface() could still take ownership.

Reply:
Should have added a comment at the time I guess 😅  Maybe I'm remembering something wrong